pkg/applicationserver/applicationserver.go

```diff
@@ -101,23 +102,23 @@ func New(c *component.Component, conf *Config) (as *ApplicationServer, err error
 
 	ctx := log.NewContextWithField(c.Context(), "namespace", "applicationserver")
 
+	baseConf := c.GetBaseConfig(ctx)
+
 	var interopCl InteropClient
 	if !conf.Interop.IsZero() {
-		var fallbackTLS *tls.Config
-		cTLS, err := c.GetTLSClientConfig(ctx)
-		if err != nil {
-			log.FromContext(ctx).WithError(err).Warn("Could not get fallback TLS config for interoperability")
-		} else {
-			fallbackTLS = cTLS
+		interopConf := conf.Interop.InteropClient
+		interopConf.GetFallbackTLSConfig = func(ctx context.Context) (*tls.Config, error) {
```

Comment:
Can't you just write

```go
interopConf.GetFallbackTLSConfig = c.GetTLSClientConfig
```

Reply:
Before rebase `GetTLSConfig` was used, which had a different signature